src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,566 @@
+    if not os.path.isdir(slc_dir):
+        err_str = f'{slc_dir} SLC directory does not exist'
+        error.log(err_str)
+        raise FileNotFoundError(err_str)
+
+    if not os.path.isfile(dem_file):
+        err_str = f'{dem_file} DEM file does not exists'
+        error.log(err_str)
+        raise FileNotFoundError(err_str)
+
+    # Create directory for runfiles
+    run_dir = f'{work_dir}/run_files'
+    os.makedirs(run_dir, exist_ok=True)
+
+    # Check if orbit are provided, if Not download
+    if orbit_dir is None:
+        info.log('Orbit directory not assigned. Download orbit ephemerides')
+
+        # Create orbit dir and check internet connection
+        orbit_dir = f'{work_dir}/orbits'
+        os.makedirs(orbit_dir, exist_ok=True)
+        check_internet_connection()
+
+        # List all zip file and extract info
+        zip_file_list = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+
+        for zip_file in zip_file_list:
+            sensor_id, _, start_datetime, \
+            end_datetime, _ = parse_safe_filename(zip_file)
+
+            # Find precise orbits first
+            orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                        end_datetime, 'AUX_POEORB')
+            # If orbit_dict is empty, precise orbits have not been found. Find restituted orbits instead
+            if orbit_dict == None:
+                orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                            end_datetime, 'AUX_RESORB')
+            # Download the orbit file
+            download_orbit(orbit_dir, orbit_dict['orbit_url'])
+
+    # Generate burst map and prune it if a list of burst ID is provided
+    zip_file = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+    burst_map = generate_burst_map(zip_file, orbit_dir, x_spac, y_spac, epsg)
+
+    # Identify burst IDs common across the stack and remove from the dataframe
+    # burst IDs that are not in common
+    common_ids = get_common_burst_ids(burst_map)
+    burst_map = prune_dataframe(burst_map, 'burst_id', common_ids)
+
+    # If user selects burst IDs to process, prune unnecessary bursts
+    if burst_id is not None:
+        burst_map = prune_dataframe(burst_map, 'burst_id', burst_id)
+
+    # Select only dates between start and end
+    if start_date is not None:
+        burst_map = burst_map[burst_map['date'] >= start_date]
+    if end_date is not None:
+        burst_map = burst_map[burst_map['date'] <= end_date]
+
+    # Exclude some dates if the user requires it
+    if exclude_dates is not None:
+        burst_map = prune_dataframe(burst_map, 'date', exclude_dates)
+
+    # Ready to geocode bursts
+    for safe in zip_file:
+        i_subswath = [1, 2, 3]
+        orbit_path = get_orbit_file_from_list(safe,
+                                              glob.glob(f'{orbit_dir}/S1*'))
+        for subswath in i_subswath:
+            bursts = load_bursts(safe, orbit_path, subswath)
```

Comment:
If [`Sentinel1BurstSlc`](https://github.com/opera-adt/s1-reader/blob/e642bd645aea820e03a3497df9de1c647cb71f0a/src/s1reader/s1_burst_slc.py#L127) objects are stored in `burst_map` then they don't need to reloaded here.

If my suggestions to filter in `generate_burst_map` are accepted, then that should keep the `burst_map` object not too big?

Reply:
This suggestions is implemented in https://github.com/opera-adt/COMPASS/pull/53 , and it means that the script run time is cut in half (since 95% or more of the script time is spent loading, and every zip was loaded twice)